### DISCUSSION TITLE
Ideas for Overview (Ctrl-W)

### DISCUSSION THREAD

**Discussion by nakermann1973** [mzur/gnome-shell-wsmatrix#202] [Ideas]
1. Is it possible to drag-drop windows between workspaces in the overview?
2. I am using dual monitor, with the setting "Multi-monitor: Workspaces on all displays".  When accessing the overview, I can only switch workspaces when I use the mouse in my Right-hand (non-primary) monitor.  If I move the mouse on the Primary monitor, the workspaces behind do not switch
3. When opening overview, I would prefer the active workspace to only change when I click on it, rather than just when I move the mouse.
4. When opening overview on my dual-monitor setup, I see one grid on each monitor as follows (each number represents a workspace).

```
1 2 3      1 2 3
4 5 6      4 5 6
7 8 9      7 8 9
```

I would prefer a single grid, with the matching workspaces from each monitor shown directly together - something like

```
11 22 33
44 55 66
77 88 99
```

**Comment by mzur** [comment]
> Is it possible to drag-drop windows between workspaces in the overview?

No but it is discussed in #60.

> I am using dual monitor, with the setting "Multi-monitor: Workspaces on all displays". When accessing the overview, I can only switch workspaces when I use the mouse in my Right-hand (non-primary) monitor. If I move the mouse on the Primary monitor, the workspaces behind do not switch

Confirmed. This can be further discussed in #224.

> When opening overview, I would prefer the active workspace to only change when I click on it, rather than just when I move the mouse.

This was discussed and fixed at some point, I think. Could you please check if the behavior is still the same?

> When opening overview on my dual-monitor setup, I see one grid on each monitor as follows (each number represents a workspace).

That would require significant work, I guess. Feel free to contribute this feature :wink:

**Comment by xiil34f** [comment]
> Is it possible to drag-drop windows between workspaces in the overview?

A temporary workaround I found for this was to assign a shortcut to 'toggle window on all workspaces', toggle all the windows you want to move, go to the workspace you want the windows moved to, and then toggle them off.

It's kinda janky but hopefully it helps someone until the feature gets implemented.

**Reply by mzur** [reply]
You can also use the "move to workspace" shortcut ([see wiki](https://github.com/mzur/gnome-shell-wsmatrix/wiki/Custom-keyboard-shortcuts#available-shortcuts-provided-by-gnome-selection)) multiple times.
